src/vs/workbench/contrib/chat/browser/chatSetup.ts

```diff
@@ -1197,7 +1197,25 @@ export class ChatSetupContribution extends Disposable implements IWorkbenchContr
 				const params = new URLSearchParams(url.query);
 				this.telemetryService.publicLog2<WorkbenchActionExecutedEvent, WorkbenchActionExecutedClassification>('workbenchActionExecuted', { id: CHAT_SETUP_ACTION_ID, from: 'url', detail: params.get('referrer') ?? undefined });
 
-				await this.commandService.executeCommand(CHAT_SETUP_ACTION_ID, validateChatMode(params.get('mode')));
+				const modeParam = params.get('mode');
+				let modeToUse: ChatModeKind | string | undefined;
+				if (modeParam) {
+					const chatModeService = this.instantiationService.invokeFunction(accessor => accessor.get(IChatModeService));
```

Comment:
Actually it would be more idiomatic to inject IChatModeService to this class (same place `this.instantiationService` comes from) rather than grab it dynamically here